jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -144,7 +276,34 @@ default boolean hasRemaining()
     }
 
     /**
-     * @return the {@code ByteBuffer} capacity
+     * @return whether the {@code ByteBuffer} has remaining bytes left for reading
+     */
+    default boolean isEmpty()
+    {
+        return !hasRemaining();
+    }
+
+    /**
+     * @return the number of remaining bytes in the {@code ByteBuffer}
+     * @see #remaining()
+     */
+    default long size()
+    {
+        return remaining();
+    }
+
+    /**
+     * @return the maximum size in bytes.
+     * @see #size()
+     */
+    default long maxSize()
+    {
+        return capacity();
+    }
+
+    /**
+     * @return the capacity
+     * @see #maxSize()
```

Comment:
Isn't it the right moment to break backward compatibility and stop using `int` sizing?

Reply:
The approach I've taken is to keep int usage for the APIs that are named the same as BB.  So `int capacity()` and `int remaining()`.  This means that existing code does not break.

But there are new equivalent methods that use `long`, so `long maxSize()` and `long size()` are the equivalent of `capacity` and `remaining`.